mkosi/__init__.py

```diff
@@ -3299,32 +3299,33 @@ def run_clean(args: Args, config: Config) -> None:
         remove_build_cache = args.force > 1
         remove_package_cache = args.force > 2
 
-    with complete_step("Removing output files…"):
-        if config.output_dir_or_cwd().exists():
-            rmtree(*(p for p in config.output_dir_or_cwd().iterdir() if p.name.startswith(config.output)))
+    if (outputs := list(config.output_dir_or_cwd().glob(f"{config.output}*"))):
+        with complete_step(f"Removing output files of {config.name()} image…"):
+            rmtree(*outputs)
 
     if remove_build_cache:
         if config.cache_dir:
-            with complete_step("Removing cache entries…"):
+            with complete_step(f"Removing cache entries of {config.name()} image…"):
                 rmtree(*(p for p in cache_tree_paths(config) if p.exists()))
 
         if config.build_dir and config.build_dir.exists() and any(config.build_dir.iterdir()):
-            with complete_step("Clearing out build directory…"):
+            with complete_step(f"Clearing out build directory of {config.name()} image…"):
                 rmtree(*config.build_dir.iterdir())
 
-    if remove_package_cache:
-        if config.cache_dir and config.cache_dir.exists() and any(config.cache_dir.iterdir()):
-            with complete_step("Clearing out package cache…"):
-                rmtree(
-                    *(
-                        config.cache_dir / p / d
-                        for p in ("cache", "lib")
-                        for d in ("apt", "dnf", "libdnf5", "pacman", "zypp")
-                    ),
-                )
+    if remove_package_cache and config.cache_dir and config.cache_dir.exists() and any(config.cache_dir.iterdir()):
```

Comment:
Same here, because since rmtree calls out to rm -f, we don't need to check whether there's anything in cache_dir.

Reply:
We do so we don't log if we're not actually going to remove anything.